File: src/main/java/com/openpms/server/user/dto/UpdateBody.java
package com.openpms.server.user.dto;

import com.alibaba.fastjson2.annotation.JSONField;
import com.alibaba.fastjson2.annotation.JSONType;
import lombok.Data;

@Data
@JSONType(schema = """
        {
            "minProperties": 1
        }""")
public class UpdateBody {
    @JSONField(schema = "{'format':'email'}")
    private String email;
    private String name;
    private String avatar;
    private int status;
}

File: src/main/kotlin/com/creator/server/user/dto/UpdateBody.kt
package com.creator.server.user.dto

import com.alibaba.fastjson2.annotation.JSONField
import com.alibaba.fastjson2.annotation.JSONType

@JSONType(
    schema = """
    {
        "minProperties": 1
    }
    """
)
data class UpdateBody(
    @JSONField(schema = "{'format':'email'}")
    val email: String? = null,
    val name: String? = null,
    val avatar: String? = null,
    val status: Boolean = false
)
